posthog/exception_capture.py

```diff
@@ -0,0 +1,54 @@
+import json
+import sys
+import threading
+from typing import TYPE_CHECKING
+
+from posthog.exception_utils import single_exception_from_error_tuple
+
+if TYPE_CHECKING:
+    from posthog.client import Client
+
+
+class ExceptionCapture:
+    # TODO: Some method of avoiding internal SDK exceptions?
```

Comment:
i guess maybe something like the JS client side rate limiting too.... we don't want a tight loop with an error to generate too many messages. not blocking since it'll take a while to shake out things like that